Tests/AsyncAwait/ChatAsyncIntegrationTests.swift

```diff
@@ -636,5 +633,32 @@ class ChatAsyncIntegrationTests: BaseAsyncIntegrationTestCase {
     try await chat.removeChannelGroup(id: channelGroup.id)
   }
 
+  func testChatAsync_AddConnectionStatusListener() async throws {
+    let expectation = expectation(description: "Status Listener")
+    let channel = try await chat.createChannel(id: randomString())
+
+    let statusStreamTask = Task {
+      for await status in chat.connectionStatusStream() {
+        if status == .online {
+          expectation.fulfill()
+        } else {
+          XCTFail("Unexpected condition")
+        }
+      }
+    }
+
+    // Keeps a strong reference to the returned AsyncStream to prevent it from being deallocated. If this object is not retained,
+    // the AsyncStream will be deallocated, which would cause the behavior being tested to fail.
+    let connectResult = channel.connect()
+    debugPrint(connectResult)
+
+    addTeardownBlock { [unowned self] in
+      statusStreamTask.cancel()
```

Comment:
Shouldn't this test remove connectionStatusListener?

Reply:
No, because cancelling this task causes cancelling an underlying AutoCloseable:

https://github.com/pubnub/swift-chat-sdk/blob/master/Sources/Chat%2BAsyncAwait.swift#L762